students/A_Zach/lesson09/donor_models.py

```diff
@@ -0,0 +1,71 @@
+class Donor():
+    """Class defining information about a given donor"""
+    
+    def __init__(self, name, donation = []):
+        # set a donor name
+        self.name = name
+        # set donation
+        if type(donation) is list:
+            self.donation = donation
+        else:
+            raise TypeError("Donations must be provided in a list")
+       
+       
+    def new_donation(self, new_donation):
+        # Add a new donation    
+        if type(new_donation) is int or type(new_donation) is float:
+            self.donation.append(new_donation)
+        else:
+            raise TypeError("Donations must be a number")
+
+
+class DonorCollection():
+    """Class defining the orginization of donors and their donations"""
+    
+    def __init__(self):
+        # define a dictionary for donors and their donations
+        self.donor_list = []
+    
+    
+    def new_donor(self, donor: Donor):
+        # Add a new donor
+        self.donor_list.append(donor)
+    
+    
+    def sort_donation(self, name, donation):
+        # If donor is in list, append donation, if not, add donor and donation
+        for donor in self.donor_list:
+            if donor.name == name.title():
+                donor.new_donation(donation)
+                return
+        new_donor = Donor(name.title(), [donation])
+        self.new_donor(new_donor)
+    
+    def list_donors(self):
+        # Generate text list of donors name to be printed
+        donor_list_text = ''
+        for donor in self.donor_list:
+            donor_list_text += donor.name + "\n"   
+        return donor_list_text
```

Comment:
you could make this one line: `return "\n".join([donor.name for donor in self.donor_list])`